pandas/tests/categorical/test_constructors.py

```diff
@@ -464,6 +465,22 @@ def test_from_codes_with_categorical_categories(self):
         with pytest.raises(ValueError):
             Categorical.from_codes([0, 1], Categorical(['a', 'b', 'a']))
 
+    def test_from_codes_with_nan_code(self):
+        # GH21767
+        codes = [1, 2, np.nan]
+        categories = ['a', 'b', 'c']
+        with pytest.raises(ValueError):
+            Categorical.from_codes(codes, categories)
+
+    def test_from_codes_with_float(self):
+        # GH21767
+        codes = [1.0, 2.0, 0]
+        categories = ['a', 'b', 'c']
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
use ``tm.assert_produces_warning`` here.

you *may* also have to change some tests which inadvertantly are testing using float codes, so pls change those (or you can assert_produces_warning there too)

Reply:
Happy to change to `assert_produces_warning`

No tests display as having warnings when I run `pytest pandas/tests/categorical/test_constructors.py`

Please tell me what you want me to fix, because I see nothing obvious